### PR TITLE
Ability to define new project and create components from signal asset

### PR DESCRIPTION
**Pull request by johnclary**
Fixes [#7073](https://github.com/cityofaustin/atd-data-tech/issues/7073) and [#7075](https://github.com/cityofaustin/atd-data-tech/issues/7075)

### CODE REVIEW

File: moped-editor/src/views/projects/newProjectView/NewProjectView.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectView.js:55]: This is not future proof. Alternatives:
- Query component types from DB and use as a lookup
- ?

File: moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js:34]: If/when we want to re-use this in the components editor—we'll need to wrap some of this project create form-specific handling. tbd?

File: moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js:109]: it's hard to verify the spinner renders when testing because the Socrata request (probably cachced) is faaast :) i've seen it, I swear.

**Comment by johnclary** [comment]
Before this merges: would like to discuss the dependency on the Socrata dataset. Need to consider it's update frequency and downtime (two saturday evenings/month).

File: moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js:7]: if we change our approach to handling drawn points, this would be a dark dependency that could break.

**Comment by johnclary** [comment]
This is ready for review—no rush as I'm OOO next week.

**Comment by chiaberry** [comment]
https://jc-7073-new-project-from-signal--atd-moped-main.netlify.app/ to see it in action ✨

File: moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js
Comment (chiaberry) [inline moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js:98]: I looked to see if it was possible, but we could do a virtualized list in the autocomplete if we wanted to. I can't imagine someone scrolling forever to find their signal though.

**Review by chiaberry — APPROVED** [review]

**Review by amenity — APPROVED** [review]
I thought I'd already approved this but perhaps that's pending in a forgotten tab. In any case, this is SO slick. 😎 🙏

**Comment by johnclary** [comment]
thanks y'all! will hold off on merging until #409 can be merged into this. WIP.

File: moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js
Comment (johnclary) [reply moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js:98]: oooh cool idea—I will read up on that

**Comment by johnclary** [comment]
Folks—I had intended to ask y'all to review #409 separately, but I merged `main` into that branch to avoid some conflicts, and which turned it into a massive PR.

Now this PR includes the component editing functionality from that branch, and is quite a massive PR. I'm sorry! @sergiogcx I do think it'd be good for you to review this. Happy to pair on it as well.

File: moped-editor/src/views/projects/projectView/ProjectComponentEdit.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponentEdit.js:238]: @sergiogcx i uncovered a tricky bug here that we somehow hadn't noticed before. Subcomponent changes were not being saved properly for components which had a required subtype, because the subtype's subcomponents were being overwritten here. I have some ideas for a more elegant fix, but this gets it done :/

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectMap.js:280]: this causes the map to auto-zoom to selecting a signal during component editing. when creating a new component, the map will not zoom to the selected signal until the component is saved. 

all of this was quite tricky because we're forking the map editing behavior specifically for signal components.

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectMap.js:461]: these click and hover events must be disabled for signal components because we're not rendering those interactive layers. we could maybe avoid some of this condition checking by refactoring the click/hover handlers.

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectMap.js:492]: And here's where we filter out the CTN clickable map layers. We are forcing the behavior that if a signal asset is selected only the point geometry from Socrata data can be stored.  If the user wants to select a signal asset but use a different geometry, this should be handled upstream by editing the signal record in the Data Tracker.

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js
Comment (johnclary) [inline moped-editor/src/views/projects/newProjectView/NewProjectMap.js:544]: Similar to above, we don't want users hand-drawing geometries for signal components.

File: moped-editor/src/views/projects/projectView/ProjectComponentSubcomponents.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponentSubcomponents.js:106]: fixes an MUI linter warning that was throwing on component init

File: moped-editor/src/utils/socrataHelpers.js
Comment (sergiogcx) [inline moped-editor/src/utils/socrataHelpers.js:29]: I live this custom hook and how you run console.error only if there is an error. Will borrow this line in the future! 👍

File: moped-editor/src/utils/mapHelpers.js
Comment (sergiogcx) [inline moped-editor/src/utils/mapHelpers.js:1077]: No action required on here, but I wonder if `signal.id` could potentially throw a WSOD in some unlikely scenario where (signal) is a value and not an object, or it is an object but does not have the "id" key. I may be mistaken, but I wonder if the use of conditional chaining is warranted in this case.

File: moped-editor/src/views/projects/newProjectView/NewProjectMap.js
Comment (sergiogcx) [reply moped-editor/src/views/projects/newProjectView/NewProjectMap.js:461]: I was wondering what would happen if a component changes type from signal to something else, but I think the code you have written should handle it just fine.

File: moped-editor/src/views/projects/newProjectView/NewProjectView.js
Comment (sergiogcx) [reply moped-editor/src/views/projects/newProjectView/NewProjectView.js:58]: ➕ Agreed. This will almost certainly have to change to a DB lookup, we are creating/editing types in the future that will affect their name and index id.

File: moped-editor/src/views/projects/newProjectView/NewProjectView.js
Comment (sergiogcx) [inline moped-editor/src/views/projects/newProjectView/NewProjectView.js:66]: Typo ☝️

File: moped-editor/src/views/projects/projectView/ProjectComponentEdit.js
Comment (sergiogcx) [reply moped-editor/src/views/projects/projectView/ProjectComponentEdit.js:238]: @johnclary Ah, thanks for finding this bug. It makes sense, and your fix is elegant as-is imo. Happy to hear the other ideas you have.

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:56]: These need to be put in a different file I think? Otherwise, it becomes an anti-pattern 🤷

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:123]: Maybe options need to be filtered to have a length of at least one character?

**Review by sergiogcx — APPROVED** [review]
This is amazing that we are integrating the map with socrata! 🗺️

**Review by mateoclarke — APPROVED** [review]
🚢 💨 ✨ 

monsters-inc-traffic-lights

File: moped-editor/src/utils/mapHelpers.js
Comment (johnclary) [reply moped-editor/src/utils/mapHelpers.js:1077]: you're right. should never happen, but it's risky to rely on perfect data from socrata 100% of the time. i'll follow your suggestion.

File: moped-editor/src/views/projects/newProjectView/NewProjectView.js
Comment (johnclary) [reply moped-editor/src/views/projects/newProjectView/NewProjectView.js:58]: 👍 thanks for the nudge. i'll re-use a query from ProjectComponents/

File: moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js
Comment (johnclary) [reply moped-editor/src/views/projects/newProjectView/SignalAutocomplete.js:43]: yes—that's exactly what's causing it. i spent a lot of time fussing with the `value` prop setter of MUI autocomplete but kept hitting a warning when the input value was `undefined`. 

i just retested this component and realized that specifically passing `null` as an input value will bypass the warnings. will fix!

File: moped-editor/src/views/projects/projectView/ProjectComponentEdit.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponentEdit.js:238]: in that case I'll leave it as-is ;) glad i was able to get familiar with this reducer—it's awesome!

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:56]: @sergiogcx—to be clear, you're suggesting to move both of these hooks outside the component?  i'm happy to do that and am inclined to make a `signalComponentHelpers` util to avoid bloating the mapHelpers any more.

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:123]: i over-engineered this because i didn't understand the normal MUI autocomplete behavior. will refactor.

File: moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js
Comment (sergiogcx) [reply moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js:56]: Ah, yes mapHelpers is way too bloated. signalComponentHelpers would be very helpful. 🙏

**Comment by johnclary** [comment]
I created [#7136](https://github.com/cityofaustin/atd-data-tech/issues/7136) to complete these follow-up tasks

**Pull request merged**
